hicommonwealth/commonwealth
# CAT thread links

## Link to Issue
Closes: #5040 

## Description of Changes
- Adds the ability to link a CAT for display in sidebar or in-line on threads 
- Adds the display option configuration to enable the sidebar vs inline
- Introduces a new `view_template` style for showing in forms and cards
<img width="362" alt="image" src="https://github.com/hicommonwealth/commonwealth/assets/36428666/68c37745-5135-424f-8600-a3e70748f495">


## Test Plan
- with CAT flag on create a thread in a community with active CATs
- add the CAT via the card and modal
- ensure it displays in sidebar by default
- test toggle of sidebar/default 
- attempt a transaction and ensure metamask pops up
- ensure regular template UX is not changed

## Other Considerations

- for context the UX/usefulness of this feature is for threads that point to a specific contract action(ie vote/propose/delegate). This allows the ability to create on-chain interactive threads with two display options 
- currently uses the CAT feature flag as its assumed this feature can be enabled if CAT is enabled. Can be updated to more granular flag if necessary
